src/WebJobs.Script/Extensions/IDictionaryExtensions.cs

```diff
@@ -19,5 +21,35 @@ public static class IDictionaryExtensions
                 dictionary[pair.Key] = pair.Value;
             }
         }
+
+        public static bool TryGetValue<TValue>(this IDictionary<string, object> obj, string name, out TValue value, bool ignoreCase = false)
+        {
+            value = default(TValue);
+
+            // first try a default case sensitive match (dictionary might contain duplicate
+            // keys varying by case only)
+            object tempValue = null;
+            bool found = obj.TryGetValue(name, out tempValue);
+
+            if (!found && ignoreCase)
+            {
+                // try a case insensitive match - first match wins
```

Comment:
I agree that would catch most cases, but I don't think this is a big perf issue really. I'd rather not get in the business of checking a reduced set of permutations (e.g. Key, key, KEY, Key)